src/main/java/kitchenpos/menu/application/MenuService.java

```diff
@@ -0,0 +1,64 @@
+package kitchenpos.menu.application;
+
+import kitchenpos.menu.domain.Menu;
+import kitchenpos.menu.domain.MenuRepository;
+import kitchenpos.menu.dto.MenuProductRequest;
+import kitchenpos.menu.dto.MenuRequest;
+import kitchenpos.menu.dto.MenuResponse;
+import kitchenpos.menugroup.domain.MenuGroup;
+import kitchenpos.menugroup.domain.MenuGroupRepository;
+import kitchenpos.product.domain.Product;
+import kitchenpos.product.domain.ProductRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class MenuService {
+    private final MenuRepository menuRepository;
+    private final MenuGroupRepository menuGroupRepository;
+    private final ProductRepository productRepository;
+
+    public MenuService(
+            final MenuRepository menuRepository,
+            final MenuGroupRepository menuGroupRepository,
+            final ProductRepository productRepository
+    ) {
+        this.menuRepository = menuRepository;
+        this.menuGroupRepository = menuGroupRepository;
+        this.productRepository = productRepository;
+    }
+
+    @Transactional
+    public MenuResponse create(final MenuRequest menuRequest) {
+        MenuGroup menuGroup = menuGroupRepository.findById(menuRequest.getMenuGroupId())
+                .orElseThrow(() -> {throw new IllegalArgumentException("메뉴그룹이 없습니다."); });
+
+        List<Product> products = productRepository.findAllByIdIn(
+                menuRequest.getMenuProducts().stream()
+                        .map(MenuProductRequest::getProductId)
+                        .collect(Collectors.toList())
+        );
+
+        Menu menu = Menu.createMenu(
+                menuRequest.getName(),
+                menuRequest.getPrice(),
+                menuGroup,
+                menuRequest.createMenuProducts(products)
+        );
+
+        BigDecimal sumPrice = menuRequest.getSumPrice(products);
+        menu.validateSumPrice(sumPrice);
+
+        return MenuResponse.of(menuRepository.save(menu));
+    }
+
+    public List<MenuResponse> list() {
```

Comment:
읽기 작업만 하는 경우 아래와 같이 추가해주면 어떨까요?

```suggestion
    @Transactional(readOnly = true)
    public List<MenuResponse> list() {
```

붙이지 않아도 로직 구현에 크게 상관은 없지만 드라이버에 따라 Writer, Reader 부하 분산을 할 수 있어서 저는 작은 규모의 프로젝트여도 습관처럼 붙이고 있습니다.

Reply:
오! 맞어.. 이런건 습관이 들어야 할거 같네요. 정말 꿀리뷰네요. 감사합니다.